packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -350,52 +367,63 @@ export function ApplicationBoard({ secondEntries, secondEntryChainTabs }: MaskAp
     ]
 
     return (
-        <>
-            {secondEntryChainTabs?.length ? (
-                <div className={classes.abstractTabWrapper}>
-                    <NetworkTab
-                        chainId={chainId}
-                        setChainId={setChainId}
-                        classes={classes}
-                        chains={secondEntryChainTabs}
+        <TargetChainIdContext.Provider>
+            <AllProviderTradeContext.Provider>
```

Comment:
I see you move these context providers from `packages/mask/src/plugins/Trader/SNSAdaptor/trader/TraderDialog.tsx` here, do some of your components rely on them too? If so, you can wrap them with them too rather than wrap a high order component, and leave the providers in `TraderDialog` to make the code compact:


```tsx
 <TargetChainIdContext.Provider>
            <AllProviderTradeContext.Provider>
			<YourComponentThatRelysTradeContext />
            </AllProviderTradeContext.Provider>
 </TargetChainIdContext.Provider>
````